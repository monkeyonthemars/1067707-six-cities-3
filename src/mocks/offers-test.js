const CITY = {
  PARIS: {
    coordinates: [48.904715, 2.359015],
    name: `PARIS`
  },
  COLOGNE: {
    coordinates: [50.935172, 6.953100],
    name: `COLOGNE`
  },
  BRUSSELS: {
    coordinates: [50.850345, 4.351720],
    name: `BRUSSELS`
  },
  AMSTERDAM: {
    coordinates: [52.38334, 4.8],
    name: `AMSTERDAM`
  },
  HAMBURG: {
    coordinates: [53.551085, 9.993681],
    name: `HAMBURG`
  },
  DUSSELDORF: {
    coordinates: [51.227740, 6.773455],
    name: `DUSSELDORF`
  }
};

const cities = [
  {
    coordinates: [48.904715, 2.359015],
    name: `PARIS`
  },
  {
    coordinates: [50.935172, 6.953100],
    name: `COLOGNE`
  },
  {
    coordinates: [50.850345, 4.351720],
    name: `BRUSSELS`
  },
  {
    coordinates: [52.38333, 4.9],
    name: `AMSTERDAM`
  },
  {
    coordinates: [53.551085, 9.993681],
    name: `HAMBURG`
  },
  {
    coordinates: [51.227740, 6.773455],
    name: `DUSSELDORF`
  }
];

const currentCity = {
  coordinates: [48.904715, 2.359015],
  name: `PARIS`
};

const rentalOffers = [
  {
    id: 0,
    mark: `Premium`,
    image: `apartment-02.jpg`,
    priceValue: 120,
    priceText: `night`,
    isBookmark: false,
    rating: 100,
    title: `Luxury Tiny House, flowerbulb region nearby A'dam`,
    type: `Apartment`,
    coordinates: [52.4009553943508, 4.86309666406198],
    city: CITY.PARIS
  }, {
    id: 1,
    mark: ``,
    image: `apartment-01.jpg`,
    priceValue: 50,
    priceText: `night`,
    isBookmark: true,
    rating: 60,
    title: `Houseboat on IJsselmeer in Amsterdam`,
    type: `Shared room`,
    coordinates: [52.399553943508, 4.87309666406198],
    city: CITY.PARIS
  }, {
    id: 2,
    mark: `Premium`,
    image: `apartment-03.jpg`,
    priceValue: 151,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Amstel Rembrandtplein`,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.909309666406198],
    city: CITY.COLOGNE
  }, {
    id: 3,
    mark: ``,
    image: `room.jpg`,
    priceValue: 32,
    priceText: `night`,
    isBookmark: true,
    rating: 100,
    title: `Room with a View on the park.`,
    type: `Private room`,
    coordinates: [52.3709553943508, 4.919309666406198],
    city: CITY.COLOGNE
  }, {
    id: 4,
    mark: `Premium`,
    image: `apartment-02.jpg`,
    priceValue: 120,
    priceText: `night`,
    isBookmark: false,
    rating: 100,
    title: `Luxury Tiny House, flowerbulb region nearby A'dam`,
    type: `Apartment`,
    coordinates: [52.4009553943508, 4.86309666406198],
    city: CITY.BRUSSELS
  }, {
    id: 5,
    mark: ``,
    image: `apartment-01.jpg`,
    priceValue: 50,
    priceText: `night`,
    isBookmark: true,
    rating: 60,
    title: `Houseboat on IJsselmeer in Amsterdam`,
    type: `Shared room`,
    coordinates: [52.399553943508, 4.87309666406198],
    city: CITY.AMSTERDAM
  }, {
    id: 6,
    mark: `Premium`,
    image: `apartment-03.jpg`,
    priceValue: 151,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Amstel Rembrandtplein`,
    type: `Apartment`,
    coordinates: [52.3809553943508, 4.909309666406198],
    city: CITY.HAMBURG
  }, {
    id: 7,
    mark: ``,
    image: `room.jpg`,
    priceValue: 32,
    priceText: `night`,
    isBookmark: true,
    rating: 100,
    title: `Room with a View on the park.`,
    type: `Private room`,
    coordinates: [52.3709553943508, 4.919309666406198],
    city: CITY.DUSSELDORF
  }
];

export default {cities, rentalOffers, currentCity};
