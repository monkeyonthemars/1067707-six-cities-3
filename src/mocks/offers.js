const CITY = {
  PARIS: {
    coordinates: [48.904716, 2.359014],
    name: `PARIS`
  },
  COLOGNE: {
    coordinates: [50.935173, 6.953101],
    name: `COLOGNE`
  },
  BRUSSELS: {
    coordinates: [50.850346, 4.351721],
    name: `BRUSSELS`
  },
  AMSTERDAM: {
    coordinates: [52.38333, 4.9],
    name: `AMSTERDAM`
  },
  HAMBURG: {
    coordinates: [53.551086, 9.993682],
    name: `HAMBURG`
  },
  DUSSELDORF: {
    coordinates: [51.227741, 6.773456],
    name: `DUSSELDORF`
  }
};

export default [
  {
    id: 0,
    mark: `Premium`,
    image: `apartment-01.jpg`,
    priceValue: 120,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [52.3909553943508, 4.85309666406198],
    city: CITY.AMSTERDAM
  }, {
    id: 1,
    mark: ``,
    image: `room.jpg`,
    priceValue: 80,
    priceText: `night`,
    isBookmark: true,
    rating: 80,
    title: `Wood and stone place`,
    type: `Private room`,
    coordinates: [48.914716, 2.369014],
    city: CITY.PARIS
  }, {
    id: 2,
    mark: ``,
    image: `apartment-02.jpg`,
    priceValue: 132,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    coordinates: [48.924716, 2.379014],
    city: CITY.PARIS
  }, {
    id: 3,
    mark: `Premium`,
    image: `apartment-03.jpg`,
    priceValue: 180,
    priceText: `night`,
    isBookmark: false,
    rating: 100,
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    coordinates: [53.541086, 9.983682],
    city: CITY.HAMBURG
  }, {
    id: 4,
    mark: `Premium`,
    image: `apartment-01.jpg`,
    priceValue: 120,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Beautiful & luxurious apartment at great location`,
    type: `Apartment`,
    coordinates: [50.945173, 6.943101],
    city: CITY.COLOGNE
  }, {
    id: 5,
    mark: ``,
    image: `room.jpg`,
    priceValue: 80,
    priceText: `night`,
    isBookmark: true,
    rating: 80,
    title: `Wood and stone place`,
    type: `Private room`,
    coordinates: [50.860346, 4.341721],
    city: CITY.BRUSSELS
  }, {
    id: 6,
    mark: ``,
    image: `apartment-02.jpg`,
    priceValue: 132,
    priceText: `night`,
    isBookmark: false,
    rating: 80,
    title: `Canal View Prinsengracht`,
    type: `Apartment`,
    coordinates: [51.327741, 6.783456],
    city: CITY.DUSSELDORF
  }, {
    id: 7,
    mark: `Premium`,
    image: `apartment-03.jpg`,
    priceValue: 180,
    priceText: `night`,
    isBookmark: false,
    rating: 100,
    title: `Nice, cozy, warm big bed apartment`,
    type: `Apartment`,
    coordinates: [51.237741, 6.753456],
    city: CITY.DUSSELDORF
  }
];
