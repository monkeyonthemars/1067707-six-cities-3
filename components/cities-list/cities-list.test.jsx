import React from 'react';
import renderer from 'react-test-renderer';
import CitiesList from './cities-list.jsx';
import mocks from '../../src/mocks/offers-test.js';

const cityClickHandler = () => {};

it(`<CitiesList /> cities-list render correctly`, () => {
  const tree = renderer.create(
      <CitiesList
        cities={mocks.cities}
        currentCity={mocks.currentCity}
        onCityClick={cityClickHandler}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
