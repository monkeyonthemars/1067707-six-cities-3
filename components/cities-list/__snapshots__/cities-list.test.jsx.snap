// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CitiesList /> cities-list render correctly 1`] = `
Array [
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item tabs__item--active"
      href="#"
      onClick={[Function]}
    >
      <span>
        PARIS
      </span>
    </a>
  </li>,
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        COLOGNE
      </span>
    </a>
  </li>,
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        BRUSSELS
      </span>
    </a>
  </li>,
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        AMSTERDAM
      </span>
    </a>
  </li>,
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        HAMBURG
      </span>
    </a>
  </li>,
  <li
    className="locations__item"
  >
    <a
      className="locations__item-link tabs__item"
      href="#"
      onClick={[Function]}
    >
      <span>
        DUSSELDORF
      </span>
    </a>
  </li>,
]
`;
